kolena/_experimental/dataset/_evaluation.py

```diff
@@ -0,0 +1,225 @@
+    validate_batch_size(batch_size)
+    init_request = LoadInferencesRequest(dataset=dataset, model=model, batch_size=batch_size)
+    yield from _BatchedLoader.iter_data(
+        init_request=init_request,
+        endpoint_path=Path.LOAD_INFERENCES.value,
+        df_class=None,
+        endpoint_api_version=API_V2,
+    )
+
+
+def _iter_metrics_raw(dataset: str, model: str, batch_size: int) -> Iterator[pd.DataFrame]:
+    validate_batch_size(batch_size)
+    init_request = LoadMetricsRequest(dataset=dataset, model=model, batch_size=batch_size)
+    yield from _BatchedLoader.iter_data(
+        init_request=init_request,
+        endpoint_path=EvaluationPath.LOAD_METRICS.value,
+        df_class=None,
+        endpoint_api_version=API_V2,
+    )
+
+
+def _fetch_inferences(dataset: str, model: str) -> pd.DataFrame:
+    df_inference_batch = list(_iter_inference_raw(dataset, model, batch_size=BatchSize.LOAD_RECORDS))
+    return (
+        pd.concat(df_inference_batch)
+        if df_inference_batch
+        else pd.DataFrame(
+            columns=["datapoint_id", COL_DATAPOINT, "inference_id", COL_INFERENCE],
+        )
+    )
+
+
+def fetch_inferences(dataset: str, model: str) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    df_data = _fetch_inferences(dataset, model)
+    df_datapoints = _to_deserialized_dataframe(df_data, column=COL_DATAPOINT)
+    df_inferences = _to_deserialized_dataframe(df_data, column=COL_INFERENCE)
+
+    return df_datapoints, df_inferences
+
+
+def _upload_inferences(model: str, df: pd.DataFrame) -> None:
+    load_uuid = init_upload().uuid
+    upload_data_frame(
+        df=df[["id", "inference"]],
+        batch_size=BatchSize.UPLOAD_RECORDS.value,
+        load_uuid=load_uuid,
+    )
+    request = UploadInferencesRequest(model=model, uuid=load_uuid)
+    response = krequests.post(Path.UPLOAD_INFERENCES, json=asdict(request))
+    krequests.raise_for_status(response)
+
+
+def _upload_metrics(
+    df: pd.DataFrame,
+    all_metrics: List[Tuple[Optional[TYPE_EVALUATION_CONFIG], pd.DataFrame]],
+) -> int:
+    df_metrics_by_eval = []
+    for eval_config, df_metrics in all_metrics:
+        df_metrics_eval = _to_serialized_dataframe(df_metrics, column=COL_METRICS)
+        df_metrics_eval["eval_config"] = json.dumps(eval_config._to_dict())
+        df_metrics_by_eval.append(pd.concat([df["inference_id"], df_metrics_eval], axis=1))
+    df_metrics = (
+        pd.concat(df_metrics_by_eval, ignore_index=True)
+        if df_metrics_by_eval
+        else pd.DataFrame(
+            columns=["inference_id", COL_METRICS],
+        )
+    )
+
+    load_uuid = init_upload().uuid
+    upload_data_frame(df=df_metrics, batch_size=BatchSize.UPLOAD_RECORDS.value, load_uuid=load_uuid)
+    request = UploadMetricsRequest(uuid=load_uuid)
+    response = krequests.post(EvaluationPath.UPLOAD_METRICS, json=asdict(request))
+    krequests.raise_for_status(response)
+    return len(df_metrics)
```

Comment:
Can we split this part into a separate function like what we had for _upload_inference so it is more reusable? The input df_metrics will be a dataframe with three columns: `inference_id`, `metrics` and `eval_config`. The logic that links inference id to the metrics relies on the order, I would have custom code to link them in galleon, and wish to use a function in the sdk to upload the dataframe.

Reply:
i think maybe we can use public method `test` on galleon for ingestion with custom `infer` or `metrics` functions.

Reply:
Feels more like a hack to me, for inference upload, we send in the function to link inference with datapoint as `infer`, and for metrics upload we send in the function to link metrics result with inference as `eval`. This would not be the most readable but it should work.

Reply:
Any reason we want to force folks to always go through the test function?

Reply:
the `test` interface is the public interface users will interactive, it has all of the functionality. 

we will just rewrite all of the `test` logic in a different way on galleon if we don't use this func in my view.

Reply:
just [abstract _process_metrics out of _upload_metrics](https://github.com/kolenaIO/kolena/pull/252/commits/6df2db663c2e8b821987de16c6a45fb88d9fa60c)